libexec/bats-core/bats-exec-test

```diff
@@ -40,24 +40,27 @@ fi
 BATS_TEST_DIRNAME="${BATS_TEST_FILENAME%/*}"
 BATS_TEST_NAMES=()
 
+# Shorthand for source-ing files relative to the BATS_TEST_DIRNAME,
+# optionally with a .bash suffix appended. If the argument doesn't
+# resolve relative to BATS_TEST_DIRNAME it is sourced as-is.
 load() {
-  local name="$1"
-  local filename
+  local file="${1:?name}"
 
-  if [[ "${name:0:1}" == '/' ]]; then
-    filename="${name}"
-  else
-    filename="$BATS_TEST_DIRNAME/${name}.bash"
+  if [[ -f "${BATS_TEST_DIRNAME}/${file}" ]]; then
+    file="${BATS_TEST_DIRNAME}/${file}"
+  elif [[ -f "${BATS_TEST_DIRNAME}/${file}.bash" ]]; then
```

Comment:
I'm wondering now if preserving existing behaviour means searching for the `.bash` version first. In the (perhaps unlikely!) event of both `my_include` and `my_include.bash` existing, would the current implementation default to `my_include` instead of the previous behaviour of `my_include.bash`? Might be easiest-verified with a dedicated test case.

Reply:
I've added a test for the ambiguous behavior, thanks for the suggestion. Given that this change will be held until the 1.3 release how do you feel about keeping the current proposed ordering (exact-match is preferred to `.bash`-suffixed)? If `.bash`-suffixed is preferred `ambiguous` is not reachable, whereas exact-match allows both `ambiguous` and `ambiguous.bash` to be loaded.

If you'd prefer to keep `.bash`-suffix prioritized LMK and I'll flip it and update the test.

---

WRT `${1:?name}`, it is the error that gets reported, and looking at it again I think the default error message is preferable so I've removed `name`. You can experiment with this syntax, try `echo ${foo:?}`, see also [Shell Parameter Expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html).

Reply:
I would prefer the `.bash`-suffixed please, it's an edge case and a minor nit but I worry about breaking any backwards compatibility :pray: Then we should be good to go : ]

Reply:
Sounds good; I left a TODO to consider revisiting this in the future (e.g. if there's a major release).